cli/standalone.rs

```diff
@@ -1,154 +1,14 @@
 
 const MAGIC_TRAILER: &[u8; 8] = b"d3n0l4nd";
 
-/// This function will try to run this binary as a standalone binary
-/// produced by `deno compile`. It determines if this is a stanalone
-/// binary by checking for the magic trailer string `D3N0` at EOF-12.
-/// After the magic trailer is a u64 pointer to the start of the JS
-/// file embedded in the binary. This file is read, and run. If no
-/// magic trailer is present, this function exits with Ok(()).
-pub fn try_run_standalone_binary(args: Vec<String>) -> Result<(), AnyError> {
-  let current_exe_path = current_exe()?;
-
-  let mut current_exe = File::open(current_exe_path)?;
-  let trailer_pos = current_exe.seek(SeekFrom::End(-16))?;
-  let mut trailer = [0; 16];
-  current_exe.read_exact(&mut trailer)?;
-  let (magic_trailer, bundle_pos_arr) = trailer.split_at(8);
-  if magic_trailer == MAGIC_TRAILER {
-    let bundle_pos_arr: &[u8; 8] = bundle_pos_arr.try_into()?;
-    let bundle_pos = u64::from_be_bytes(*bundle_pos_arr);
-    current_exe.seek(SeekFrom::Start(bundle_pos))?;
-
-    let bundle_len = trailer_pos - bundle_pos;
-    let mut bundle = String::new();
-    current_exe.take(bundle_len).read_to_string(&mut bundle)?;
-    // TODO: check amount of bytes read
-
-    if let Err(err) = tokio_util::run_basic(run(bundle, args)) {
-      eprintln!("{}: {}", colors::red_bold("error"), err.to_string());
-      std::process::exit(1);
-    }
-    std::process::exit(0);
-  } else {
-    Ok(())
-  }
-}
-
-const SPECIFIER: &str = "file://$deno$/bundle.js";
-
-struct EmbeddedModuleLoader(String);
-
-impl ModuleLoader for EmbeddedModuleLoader {
-  fn resolve(
-    &self,
-    _op_state: Rc<RefCell<OpState>>,
-    specifier: &str,
-    _referrer: &str,
-    _is_main: bool,
-  ) -> Result<ModuleSpecifier, AnyError> {
-    if specifier != SPECIFIER {
-      return Err(type_error(
-        "Self-contained binaries don't support module loading",
-      ));
-    }
-    Ok(ModuleSpecifier::resolve_url(specifier)?)
-  }
-
-  fn load(
-    &self,
-    _op_state: Rc<RefCell<OpState>>,
-    module_specifier: &ModuleSpecifier,
-    _maybe_referrer: Option<ModuleSpecifier>,
-    _is_dynamic: bool,
-  ) -> Pin<Box<deno_core::ModuleSourceFuture>> {
-    let module_specifier = module_specifier.clone();
-    let code = self.0.to_string();
-    async move {
-      if module_specifier.to_string() != SPECIFIER {
-        return Err(type_error(
-          "Self-contained binaries don't support module loading",
-        ));
-      }
-      Ok(deno_core::ModuleSource {
-        code,
-        module_url_specified: module_specifier.to_string(),
-        module_url_found: module_specifier.to_string(),
-      })
-    }
-    .boxed_local()
-  }
-}
-
-async fn run(source_code: String, args: Vec<String>) -> Result<(), AnyError> {
-  let mut flags = Flags::default();
-  flags.argv = args[1..].to_vec();
-  // TODO(lucacasonato): remove once you can specify this correctly through embedded metadata
-  flags.unstable = true;
```

Comment:
@lucacasonato I don't think we need `flags` anymore. In the PR, this is removed, the code will simply pass all the args from the cli. This also allow us not to use flags module in deno runtime binary.

Reply:
We shouldn't change that in this PR, especially since @nayeemrmn got https://github.com/denoland/deno/pull/8738 going that would address it fully.

Reply:
It seems this PR blocked by #8738 
I will continue this after that PR is merged.